src/modules/html/producer/html_producer.cpp

```diff
@@ -399,6 +471,51 @@ class html_client
         return false;
     }
 
+    bool GetAudioParameters(CefRefPtr<CefBrowser> browser, CefAudioParameters& params) override
+    {
+        params.channel_layout    = CEF_CHANNEL_LAYOUT_7_1;
+        params.sample_rate       = format_desc_.audio_sample_rate;
+        params.frames_per_buffer = format_desc_.audio_cadence[0];
+        return format_desc_.audio_cadence.size() == 1; // TODO - handle 59.94
+    }
+
+    void OnAudioStreamStarted(CefRefPtr<CefBrowser> browser, const CefAudioParameters& params, int channels) override
+    {
+        audioResampler_ = std::make_unique<ffmpeg::AudioResampler>(params.sample_rate, AV_SAMPLE_FMT_FLTP);
+    }
+    void OnAudioStreamPacket(CefRefPtr<CefBrowser> browser, const float** data, int samples, int64_t pts) override
+    {
+        if (!audioResampler_)
+            return;
+
+        auto audio       = audioResampler_->convert(samples, reinterpret_cast<const void**>(data));
+        auto audio_frame = core::mutable_frame(this, {}, std::move(audio), core::pixel_format_desc());
+
+        {
```

Comment:
This crashes on Linux. If i comment out this entire block, it runs.

Reply:
Perhaps this is from the hardcoded 8 in https://github.com/CasparCG/server/blob/72b5ef2c7ad129474f6a519cda0abe922f1e696d/src/modules/ffmpeg/util/audio_resampler.cpp#L31?  
I appear to have opened this PR a few hours before I made a change to support 16 channel audio https://github.com/CasparCG/server/commit/28c4dcd415d0a2d01b11d789ddd97e8e9dfdd178

That would explain the slowness/broken audio you are experiencing too

Reply:
I'll try that